modules/diskq/qdisk.c

```diff
@@ -427,6 +427,16 @@ _is_trying_to_read_from_unwritten_space(QDisk *self)
   return self->hdr->read_head == self->hdr->write_head;
 }
 
+static gssize
+_read_record_length_from_disk(QDisk *self, guint32 *record_length)
```

Comment:
optional: I was about to comment the lack of this (6aca8bf3961ae83a2d68ebf80bbcd0ab0b0ca397) refactor here, when I found it. Squash them together?